genesis/camino_genesis_test.go

```diff
@@ -0,0 +1,290 @@
+		},
+		"valid": {
+			args: args{
+				config: &KopernikusConfig,
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := validateCaminoConfig(tt.args.config)
+
+			if tt.err != nil {
+				require.ErrorContains(t, err, tt.err.Error())
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestBuildCaminoGenesis(t *testing.T) {
+	ctx := snow.DefaultContextTest()
+	rewards := reward.NewCalculator(defaultRewardConfig)
+	ctx.NetworkID = constants.KopernikusID
+	baseDB := memdb.New()
+	db := versiondb.New(baseDB)
+	cfg := defaultConfig()
+
+	addrs := set.Set[ids.ShortID]{}
+	_, _, avaxAddrBytes, _ := address.Parse(initialAdmin)
+	avaxAddr, _ := ids.ToShortID(avaxAddrBytes)
+	addrs.Add(avaxAddr)
+	outputOwners := secp256k1fx.OutputOwners{
+		Locktime:  0,
+		Threshold: 1,
+		Addrs:     []ids.ShortID{avaxAddr},
+	}
+	type args struct {
+		config *Config
+		hrp    string
+	}
+	tests := map[string]struct {
+		args          args
+		expectedUtxos map[ids.ID]*avax.UTXO
+		err           error
+	}{
+		"success - kopernikus": {
+			args: args{
+				config: &KopernikusConfig,
+				hrp:    constants.KopernikusHRP,
+			},
+			expectedUtxos: map[ids.ID]*avax.UTXO{
+				expectedUtxoID1: generateTestUTXO(expectedUtxoID1, ctx.AVAXAssetID, 2000000000000, outputOwners, expectedUtxoID1, expectedBondTxID),
+				expectedUtxoID2: generateTestUTXO(expectedUtxoID2, ctx.AVAXAssetID, 10000000000000000, outputOwners, expectedUtxoID2, ids.Empty),
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			genesisBytes, _, err := buildCaminoGenesis(tt.args.config, tt.args.hrp)
+			if tt.err != nil {
+				require.ErrorIs(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+			s := defaultState(genesisBytes, cfg, ctx, db, rewards)
+
+			utxos, err := avax.GetAllUTXOs(s, addrs)
+			require.NoError(t, err)
+			require.Equal(t, len(tt.expectedUtxos), len(utxos))
+			for _, utxo := range utxos {
+				require.Equal(t, tt.expectedUtxos[utxo.TxID].TxID, utxo.TxID)
+				require.Equal(t, tt.expectedUtxos[utxo.TxID].Out, utxo.Out)
+			}
+
+			offers, err := s.GetAllDepositOffers()
+			require.NoError(t, err)
+
+			genesisOffers := tt.args.config.Camino.DepositOffers
+
+			for _, offer := range offers {
+				require.Equal(t, getGenesisOffer(offer.ID, genesisOffers).Start, offer.Start)
+				require.Equal(t, getGenesisOffer(offer.ID, genesisOffers).End, offer.End)
+				require.Equal(t, uint64(1), offer.MinAmount)
+				require.GreaterOrEqual(t, wrappers.IgnoreError(math.Sub(offer.End, offer.Start)), uint64(offer.MinDuration))
+				require.Equal(t, offer.MinDuration, offer.MaxDuration)
+				require.Equal(t, offer.NoRewardsPeriodDuration*2, offer.UnlockPeriodDuration)
+				require.Equal(t, deposit.OfferFlagLocked, offer.Flags)
+			}
+		})
+	}
+}
+
+func getGenesisOffer(id ids.ID, offers []genesis.DepositOffer) genesis.DepositOffer {
+	for _, do := range offers {
+		doID, _ := do.ID()
```

Comment:
I'd prefer to not ignore erros
in cases like this, avax uses `panic(err)`

Reply:
i do not find a significant reason for checking errors on logic "external" to the tests itself. in any case it will panic and disrupt the pipeline.  but whatever..

Reply:
not sure if it would panic on its own - I assume u'll just get the emptyID, and that could shown up as some error in test, but would be harder to track it
and yea, panic ins't good solution (just how avax does it), cause it will disrupt test pipline, but I guess its golden middle between usability (no error returned for test helper func, cause it shouldn't happen at all unless something is significantly broke, like not-registered in coded types inside offer; and in this test, we'r assuming offer struct is ok) and reliability (it won't do anything unexpected if error really happen)